apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2.java

```diff
@@ -449,8 +472,10 @@ private Coverage transformPartD(Beneficiary beneficiary, Profile profile) {
       addCoverageDecimalExtension(
           coverage, CcwCodebookVariable.RFRNC_YR, beneficiary.getBeneEnrollmentReferenceYear());
 
-      // Monthly Medicare-Medicaid dual eligibility codes
-      transformEntitlementDualEligibility(coverage, beneficiary);
+      if (!isProfileC4DIC) {
```

Comment:
Could this go in the `else` clause of the subsequent conditional to mirror what you have for the part C transformer? That will eliminate one extra conditional.

Reply:
I tried that but this needs to happen under the following condition
` if (beneficiary.getBeneEnrollmentReferenceYear().isPresent()) ` which in my test cases C4DIC data wasn't satisfying

Reply:
What's the reason for that check? Should the corresponding C4DIC logic also be inside that check?

Reply:
so I was testing a case that I kinda manipulated in a way to view hidden values but looks like this okay with the actual test data. I have tested both the normal and C4DIC cases and this should be okay